elastic/apm-server
# Introduce Kustomize and Tilt

## Motivation/summary

Introduce Kustomize manifests in `testing/infra/k8s`, using ECK for running Elasticsearch, Kibana, Fleet Server, and an Elastic Agent with the APM integration injected.

We use the Kustomize manifests from Tiltfile, injecting a live-reloaded custom image into the apm-server Agent pod spec.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Install Tilt: https://docs.tilt.dev/install.html. On macOS you can use Docker Desktop, on Linux you should use Kind (optionally using https://github.com/tilt-dev/ctlptl for managing the cluster.)
2. Run `tilt up`
3. Send some APM data, check that it shows up in Kibana

## Related issues

None
